notebooks/facloc-disperse-real-world.ipynb

```diff
@@ -0,0 +1,1695 @@
+    "    facility_points,geometry=geopandas.points_from_xy(\n",
+    "    facility_points.long, facility_points.lat))\n",
+    "facility_points_gdf"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The below method is used to plot the results of the solved P-Dispersion problems."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 47,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from matplotlib.patches import Patch\n",
+    "import matplotlib.lines as mlines\n",
+    "from matplotlib_scalebar.scalebar import ScaleBar\n",
+    "\n",
+    "dv_colors = [\n",
+    "    \"darkslategray\",\n",
+    "    \"lightskyblue\",\n",
+    "    \"limegreen\",\n",
+    "    \"darkgoldenrod\",\n",
+    "    \"peachpuff\",\n",
+    "    \"coral\",\n",
+    "    \"mediumvioletred\",\n",
+    "    \"blueviolet\",\n",
+    "    \"fuchsia\",\n",
+    "    \"thistle\",\n",
+    "    \"lavender\",\n",
+    "    \"saddlebrown\",\n",
+    "    \"darkcyan\",\n",
+    "    \"mediumseagreen\",\n",
+    "    \"cyan\",\n",
+    "]\n",
+    "\n",
+    "pre_colors = [\n",
+    "    \"darkcyan\",\n",
+    "    \"mediumseagreen\",\n",
+    "    \"cyan\",\n",
+    "]\n",
+    "\n",
+    "def plot_results(model, facility_points, title):\n",
+    "    fac_sites = [] \n",
+    "    \n",
+    "    for i in range(len(facility_points)):\n",
+    "        if model.fac_vars[i].varValue == 1:\n",
+    "            fac_sites.append(i)\n",
+    "    sited_facilities = range(len(fac_sites))\n",
+    "\n",
+    "    fig, ax = plt.subplots(figsize=(10, 15))\n",
+    "    legend_elements = []\n",
+    "\n",
+    "    facility_points.plot(ax=ax, color='brown', marker=\"*\", markersize=80, zorder=2)\n",
+    "    legend_elements.append(mlines.Line2D(\n",
+    "        [],\n",
+    "        [],\n",
+    "        color='brown',\n",
+    "        marker=\"*\",\n",
+    "        linewidth=0,\n",
+    "        label=f'facility sites ($n$={len(facility_points)})'\n",
+    "    ))\n",
+    "    \n",
+    "    study_area.plot(ax=ax, alpha=.5, fc=\"tan\", ec=\"k\", zorder=1)\n",
+    "    _patch = Patch(alpha=.5, fc=\"tan\", ec=\"k\", label=\"Dissolved Service Areas\")\n",
+    "    legend_elements.append(_patch)\n",
+    "\n",
+    "    if 'predefined_loc' in facility_points.columns:\n",
+    "    \n",
+    "        pre_fac = facility_points[facility_points.predefined_loc == 1]\n",
+    "        for i in range(len(pre_fac)):\n",
+    "            pre_fac.iloc[[i]].plot(ax=ax,\n",
+    "                        marker=\"*\",\n",
+    "                        markersize=200 * 3.0,\n",
+    "                        alpha=0.8,\n",
+    "                        zorder=4,\n",
+    "                        edgecolor=\"red\",\n",
+    "                        facecolor=pre_colors[i])\n",
+    "\n",
+    "        other_fac = facility_points[facility_points.predefined_loc == 0]\n",
+    "        for i in sited_facilities:\n",
```

Comment:
Aha! Gotcha! This is a super tricky thing. 
In this case: 
`fac_sites = [0, 3, 5, 6, 7, 11]`
`sited_facilities = [0, 1, 2, 3, 4, 5]`

`other_fac` contains stores sites, except predefined ones and it has 16 rows (index ranging from 0 to 15), so when you index with `sited_facilities` you are using the right indexes but in a dataset that doesn't correspond to the facilities chosen. So, it will print more stores than expected 9 in this case instead of 6. 

I recommend taking a look at this [gist](https://gist.github.com/gegen07/32f961686f4b339870a72291ae32c322) I did which makes it simpler and less repetitive. It also solves the issue of printing more facilities than expected.